src/api.v2/controllers/experimentController.js

```diff
@@ -1,41 +1,101 @@
 /* eslint-disable import/prefer-default-export */
+const _ = require('lodash');
+
 const config = require('../../config');
 
 const experiment = require('../model/experiment');
 const userAccess = require('../model/userAccess');
 
 const AccessRole = require('../../utils/enums/AccessRole');
-
-const { OK } = require('../../utils/responses');
-
 const getLogger = require('../../utils/getLogger');
+const { OK } = require('../../utils/responses');
+const { NotFoundError } = require('../../utils/responses');
 
 const logger = getLogger('[ExperimentController] - ');
 
+const getExperiment = async (req, res) => {
+  const { params: { experimentId } } = req;
+
+  const data = await experiment.getExperimentData(experimentId);
+
+  if (_.isNil(data)) {
+    throw new NotFoundError('Experiment not found');
+  }
+
+  res.json(data);
+};
+
 const createExperiment = async (req, res) => {
   const { params: { experimentId }, user, body } = req;
 
   const { name, description } = body;
 
   logger.log('Creating experiment');
 
-  await experiment.create({ id: experimentId, name, description });
+  const experimentCreateResult = await experiment.create({ id: experimentId, name, description });
+
+  if (experimentCreateResult.length === 0) {
+    throw new Error(`Experiment ${experimentId} creation failed`);
+  }
 
   logger.log('Setting up access permissions for experiment');
-  await Promise.all([
+  const userAccessCreateResults = await Promise.all([
     userAccess.create(
       { user_id: user.sub, experiment_id: experimentId, access_role: AccessRole.OWNER },
     ),
     userAccess.create(
-      { user_id: config.adminSub, experiment_id: experimentId, access_role: AccessRole.OWNER },
+      { user_id: config.adminSub, experiment_id: experimentId, access_role: AccessRole.ADMIN },
```

Comment:
with the way it is implemented right now, it will fail if you try to add two different roles for the same user and experiment, which I think makes sense.

I don't think doing them concurrently will change things except on this case you mentioned.

As a counter proposal, I think maybe I could add handling specifically for the case where `config.adminSub` === `user.sub` (only create the admin as owner, right?)

Reply:
let me know what you think about the idea

Reply:
when you say it will fail when adding two roles, you mean using this Promise.all or in general? because changing permissions should always be an overwriting operations, if you set a different role for a user that already has one, it should overwrite not fail. I'm ok with adding specific handling for the admin.